queue_services/payment-reconciliations/src/reconciliations/eft/eft_record.py

```diff
@@ -0,0 +1,91 @@
+# Copyright © 2023 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This manages the EFT Transaction record."""
+import decimal
+from datetime import datetime
+
+from reconciliations.eft.eft_base import EFTBase
+from reconciliations.eft.eft_enums import EFTConstants
+from reconciliations.eft.eft_errors import EFTError
+from reconciliations.eft.eft_parse_error import EFTParseError
+
+
+class EFTRecord(EFTBase):
+    """Defines the structure of the transaction record of a received EFT file."""
+
+    ministry_code: str
+    program_code: str
+    location_id: str
+    deposit_datetime: datetime
+    transaction_sequence: str  # optional
+    transaction_description: str
+    deposit_amount: decimal
+    currency: str  # blank = CAD, US = USD
+    exchange_adj_amount: decimal  # in cents
+    deposit_amount_cad: decimal  # Deposit amount in CAD, in cents
+    dest_bank_number: str
+    batch_number: str  # optional; specified only if posted to GL
+    jv_type: str  # I = inter, J = intra; mandatory if JV batch specified
+    jv_number: str  # mandatory if JV batch specified
+    transaction_date: datetime  # optional
+
+    def __init__(self, content: str, index: int):
+        """Return an EFT Transaction record."""
+        super().__init__(content, index)
+        self._process()
+
+    @staticmethod
+    def get_currency(currency: str) -> str:
+        """Get the appropriate exchange currency."""
+        if len(currency.strip()) == 0:  # Blank currency means CAD
+            return EFTConstants.CURRENCY_CAD.value
+
+        return currency
+
+    def _process(self):
+        """Process and validate EFT Transaction record string."""
+        # Confirm line length is valid, skip if it is not, but add to error array
+        if not self.is_valid_length():
+            self.add_error(EFTParseError(EFTError.INVALID_LINE_LENGTH))
+            return
+
+        # Confirm record type is as expected
+        self.record_type = self.content[0:1]
+        self.validate_record_type(EFTConstants.TRANSACTION_RECORD_TYPE.value)
+
+        self.ministry_code = self.content[1:3].strip()
```

Comment:
Could we include a link to the spec? Or a comment listing the spec? 

That way the next person knows where were building this from?

Reply:
Added a comment with the spec in eft_base